src/interface/meshblock_data.hpp

```diff
@@ -148,18 +146,20 @@ class MeshBlockData {
                              "Couldn't find variable '" + label + "'");
     return it->second;
   }
+  std::shared_ptr<CellVariable<T>> GetCellVarPtr(const Uid_t &uid) const {
+    return varUidMap_.at(uid);
+  }
 
   CellVariable<T> &Get(const std::string &base_name,
                        int sparse_id = InvalidSparseID) const {
     return *GetCellVarPtr(MakeVarLabel(base_name, sparse_id));
   }
-  CellVariable<T> &Get(const int index) const { return *(varVector_[index]); }
+  CellVariable<T> &Get(const Uid_t &uid) const { return *(varUidMap_.at(uid)); }
 
-  int Index(const std::string &label) noexcept {
-    for (int i = 0; i < (varVector_).size(); i++) {
-      if (!varVector_[i]->label().compare(label)) return i;
-    }
-    return -1;
+  Uid_t UniqueID(const std::string &label) noexcept {
+    auto it = varMap_.find(label);
+    if (it == varMap_.end()) return 0; // Invalid ID
```

Comment:
Is it worth promoting this magic number to a named quantity e.g. `vpack_types::invalidID`? I wouldn't necessarily be able to guess whether the right magic number is `0` or `-1` since I have to look elsewhere to know that the unique ID is unsigned.

Reply:
Done